### PR TITLE
AppArmor recorder: do not include CAP_OPT_NOAUDIT events

### PR DESCRIPTION
**Pull request by mhils**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

This ensures that we do not end up with unused (sensitive) permissions in the profile.

#### Which issue(s) this PR fixes:

None

#### Does this PR have test?

Yes.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: .typos.toml [hunk 1]
```diff
@@ -23,7 +23,7 @@ extend-ignore-identifiers-re = []
 extend-ignore-re = [
     "(?i)aks",
     "ANDed",
-    "PT_REGS_PARM3",
+    "PT_REGS_PARM",
```
Comment (ccojocar) [inline .typos.toml:26]: Is this change intend it?

File: internal/pkg/daemon/bpfrecorder/bpf/recorder.bpf.c [hunk 2]
```diff
@@ -281,6 +283,11 @@ int BPF_KPROBE(cap_capable)
         return 0;
 
     unsigned long cap = PT_REGS_PARM3(ctx);
+    unsigned long cap_opt = PT_REGS_PARM4(ctx);
+    // bpf_printk("requesting capability: cap=%i cap_opt=%i\n", cap, cap_opt);
```
Comment (ccojocar) [inline internal/pkg/daemon/bpfrecorder/bpf/recorder.bpf.c:287]: Maybe this commented line can be removed.

**Review by ccojocar — APPROVED** [review]
Thanks, lgtm

File: .typos.toml (hunk 1 above)
Comment (mhils) [reply .typos.toml:26]: Yes, the linter started to complain about `PT_REGS_PARM4`. I figured that `PT_REGS_PARM` is always ok. :)

**Comment by saschagrunert** [comment]
@mhils may I ask you to rebase?

**Comment by mhils** [comment]
Done! Still based on the OLM PR so that CI turns green. :)

**Pull request merged**